### PR TITLE
week 6 added

### LINKED ISSUE #27: Week 6 (Structures, Typeclasses)
- more type theory and structures

### PR DESCRIPTION
**Pull request by PatrickKinnear**
Aiming to merge by Monday 12:00 if possible! After that I'll be offline until Tuesday.

### CODE REVIEW

File: Formal2024/Workshops/workshop6.lean [hunk 1]
```diff
@@ -0,0 +1,272 @@
...
+#check @Point.smul
+
+def p := Point.mk 1 2
+def q := Point.mk 3 4
+
+-- Notice the order in which arguments are input to functions here.
+#eval p.add q
+#check p.add q
+
+#eval p.smul 3
```
Comment (AdrianDoM) [inline Formal2024/Workshops/workshop6.lean:89]: How does this work? My guess would be that for any function `f` defined in the `Point` namespace taking at least one `p : Point` as an argument, then `p.f` is `Point.f` with `p` as its first `Point` argument? If that is the case, should we write something about that (or maybe it's fine to just mention it during the demo).

File: Formal2024/Workshops/workshop6.lean [hunk 2]
```diff
@@ -0,0 +1,272 @@
...
+  red : ℕ
+  green : ℕ
+  blue : ℕ
+
+structure RGBPoint (α : Type*) extends Point α, RGBValue
+
+#print prefix RGBPoint
+#check RGBPoint.mk
+#check RGBPoint.toPoint
+
```
Comment (AdrianDoM) [inline Formal2024/Workshops/workshop6.lean]:
```suggestion
#check RGBPoint.toRGBValue
```

File: Formal2024/Homeworks/homework6.lean [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+    (a b : StandardTwoSimplex) : StandardTwoSimplex :=
+  sorry
+
+end -- end the `noncomputable section`
+
+end StandardTwoSimplex
+
+/-!
+# Problem 2: Classes [MIL 7.3]
+In this problem we will combine classes with what we learned about quotient types. Our goal is to prove that the quotient of a commutatuve monoid by a submonoid is again a monoid.
```
Comment (AdrianDoM) [inline Formal2024/Homeworks/homework6.lean]:
```suggestion
In this problem we will combine classes with what we learned about quotient types. Our goal is to prove that the quotient of a commutative monoid by a submonoid is again a monoid.
```

File: Formal2024/Homeworks/homework6.lean [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+import Library
+
+/-!
+# Homework 6: Structures and Classes
+References: [MIL] Mathematics in Lean.
+-/
+
+/-!
+# Problem 1: Structures [MIL 6.1]
+Here define the structure of a standard simplex and ask you to prove something about it.
```
Comment (themathqueen) [inline Formal2024/Homeworks/homework6.lean]:
```suggestion
Here we define the structure of a standard simplex and ask you to prove something about it.
```

File: Formal2024/Workshops/workshop6.lean (hunk 1 above)
Comment (PatrickKinnear) [reply Formal2024/Workshops/workshop6.lean:89]: Yes, your guess is right and I had planned to mention it in the demo. My thinking was that the sheets can get too verbose if those kinds of small details are all mentioned in detail, so my hope is that prompting the reader to think about it and me mentioning it in class is enough.

Comment (AdrianDoM) [reply Formal2024/Workshops/workshop6.lean:89]: Cool! I think it's fine to just mention it.

**Review by AdrianDoM — APPROVED** [review]
Great stuff. Thanks Patrick!

**Pull request merged**